apiserver/facades/agent/credentialvalidator/credentialvalidator.go

```diff
@@ -58,13 +58,35 @@ type CloudService interface {
 	Cloud(ctx context.Context, name string) (*cloud.Cloud, error)
 }
 
+// ModelService provides access to the model.
+type ModelService interface {
+	// WatchModelCloudCredential returns a new NotifyWatcher watching for changes that
+	// result in the cloud spec for a model changing. The changes watched for are:
+	// - updates to model cloud.
+	// - updates to model credential.
+	// - changes to the credential set on a model.
+	// The following errors can be expected:
+	// - [modelerrors.NotFound] when the model is not found.
+	WatchModelCloudCredential(ctx context.Context, modelUUID model.UUID) (corewatcher.NotifyWatcher, error)
+}
+
+// ModelInfoService provides access to the model info.
+type ModelInfoService interface {
+	// GetModelInfo returns the readonly model information for the model in
+	// question.
+	GetModelInfo(ctx context.Context) (model.ModelInfo, error)
+}
+
 type CredentialValidatorAPI struct {
 	*credentialcommon.CredentialManagerAPI
 
 	logger            logger.Logger
-	backend           StateAccessor
+	backend           credentialcommon.StateBackend
```

Comment:
Don't do this. The only method used on that interface is:
`CloudCredentialTag() (names.CloudCredentialTag, bool, error)`

We can get all the info we need off the ModelInfo struct. I model tag is just cloud name, credential owner name, credential name.  We can trivially do a little extra work here and not need this backend at all. We don't even need credential tag - all we do is convert it back to a Key anyway.

Reply:
I see thanks for the clarifications ian! Initally StateAccessor contains both credentialcommon.StateBackend and Model(), and I changed this to just  credentialcommon.StateBackend after removing Model(). I will refactor this accordingly then, thanks Ian :D